configTools/src/main/scala/com/gu/janus/policy/Iam.scala

```diff
@@ -0,0 +1,147 @@
+package com.gu.janus.policy
+
+import io.circe.syntax._
+import io.circe.{Encoder, Json, JsonObject}
+
+import scala.collection.immutable.ListMap
+
+/** This models the AWS IAM policy types and subtypes.
+  *
+  * See https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies.html
+  * for the logic of the Json encoding.
+  */
+object Iam {
+
+  sealed trait Effect
+  object Effect {
+    case object Allow extends Effect
+    case object Deny extends Effect
+
+    implicit val encoder: Encoder[Effect] = Encoder.instance {
+      case Allow => Json.fromString("Allow")
+      case Deny  => Json.fromString("Deny")
+    }
+  }
+
+  case class Action(name: String)
+  object Action {
+    implicit val encoder: Encoder[Action] = Encoder[String].contramap(_.name)
+  }
+
+  case class Resource(id: String)
+  object Resource {
+    implicit val encoder: Encoder[Resource] = Encoder[String].contramap(_.id)
+  }
+
+  case class Condition(
+      key: String,
+      typeName: String,
+      conditionValues: Seq[String]
+  )
+  object Condition {
+    implicit val encoder: Encoder[Condition] = Encoder.instance { condition =>
```

Comment:
Is this actually being used? (The same applies to the Principal below). I ask because [in the Statement serialisation](https://github.com/guardian/janus-app/pull/509/files#diff-70fa0cae455542b80f233a55798d6f87af322305b5deb9b4786cd84ade56db7eR93-R95) we duplicate this implementation rather than deferring to this serialisation.

I think either way round is fine (inlining it all into the statement serialisation, or separating the serialisation of each type into its own encoder), but I don't think we should leave unused encoders lying around so that people don't suffer the confusion of editing them without effect in the future?

Reply:
I did it like this because the encoding spec appeared to show a single condition has a different encoding to multiple conditions but you're right it's only ever encoded as part of a policy so this is redundant and confusing.

Reply:
Removed in 7d52d1e0f6c82ee3037f26cf29b4ca4a883c8f95